packages/fetchai/protocols/oef/custom_types.py

```diff
@@ -0,0 +1,142 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2020 fetchai
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains class representations corresponding to every custom type in the protocol specification."""
+
+import base64
+import pickle  # nosec
+from enum import Enum
+
+from aea.helpers.search.models import Description as BaseDescription
+from aea.helpers.search.models import Query as BaseQuery
+
+
+class Description(BaseDescription):
+    """This class represents an instance of Description."""
+
+    @classmethod
+    def encode(cls, performative, description_from_message: "Description"):
```

Comment:
Eventually, BaseDescription and BaseQuery should come with their own serizliation logic using protobuf